webdriver-spec.html

```diff
@@ -5512,6 +5522,33 @@ <h2>Element Interaction</h2>
  they will implicitly <a data-lt="scroll into view">scroll elements into view</a>
  and check that it is an <a>interactable element</a>.
 
+<p>Some <a>resettable elements</a> define their own <dfn>clear algorithm</dfn>.
+ Unlike their associated <a>reset algorithms</a>,
+ changes made to form controls as part of these algorithms
+ <em>does</em> count as changes caused by the user
+ (and thus, e.g. does cause <a><code>input</code></a> events to fire).
+ When the <a>clear algorithm</a> is invoked
+ for an element that does not define its own <a>clear algorithm</a>,
+ its <a>reset algorithm</a> must be invoked instead.
+
+<p>The <a>clear algorithm</a> for <a><code>input</code></a> elements
+ is to set the <a>dirty value flag</a> and <a>dirty checkedness flag</a> back to false,
+ set the <a>value</a> of the element to an empty string,
```

Comment:
What if the empty value doesn't pass the correctness tests for the new value?

Reply:
An empty string should always be correct for `<input>` elements, seeing as `<input value="">` is always allowed. `<input required>` may not allow the form to be _submitted_, but that’s unrelated.